lib/portlayer/network/context.go

```diff
@@ -1058,9 +1113,27 @@ func (c *Context) ContainerByAddr(addr net.IP) *Endpoint {
 
 func (c *Context) DeleteScope(name string) error {
 	defer trace.End(trace.Begin(""))
+
 	c.Lock()
 	defer c.Unlock()
 
+	var err error
+	if c.kv != nil {
+		op := trace.NewOperation(context.Background(), fmt.Sprintf("deleteing scope %s from kv", name))
+		if err = c.kv.Delete(op, fmt.Sprintf("context.scopes.%s", name)); err != nil && err != kvstore.ErrKeyNotFound {
+			return err
+		}
+	}
+
+	// FIXME: restore scope in kv store if deleteScope() fails
+	if err = c.deleteScope(name); err != nil {
```

Comment:
It could be either way; its just easier to restore the kv store data than it is to restore the scope.